apps/internal/base/base.go

```diff
@@ -262,15 +262,15 @@ func (b Client) AcquireTokenSilent(ctx context.Context, silent AcquireTokenSilen
 	result, err := AuthResultFromStorage(storageTokenResponse)
 	if err != nil {
 		if reflect.ValueOf(storageTokenResponse.RefreshToken).IsZero() {
-			return AuthResult{}, errors.New("no refresh token found")
+			return AuthResult{}, errors.New("no token found")
 		}
 
 		var cc *accesstokens.Credential
 		if silent.RequestType == accesstokens.ATConfidential {
 			cc = silent.Credential
 		}
 
-		token, err := b.Token.Refresh(ctx, silent.RequestType, b.AuthParams, cc, storageTokenResponse.RefreshToken)
```

Comment:
This is problem 1. `b.AuthParams` holds common parameters for all auth requests. Its type has a scopes field but on this instance that field always has a zero value (default scopes are handled elsewhere). `authParams` is an instance of the same type which carries overriding values for this auth request, and the scopes requested by the user application.